django/db/models/sql/compiler.py

```diff
@@ -1407,6 +1407,7 @@ def execute_sql(self, returning_fields=None):
 class SQLDeleteCompiler(SQLCompiler):
     @cached_property
     def single_alias(self):
+        self.query.get_initial_alias()
```

Comment:
This ensures `query.alias_map` and `.alias_refcount` get initialized properly. This pattern is used in other location in this file for setup purposes. Without it the `sum` returns 0 and thus this method returns `False` while it should be `True`.